### ISSUE TITLE
3/12/20 data issues still persist in next day's timeseries

### ISSUE THREAD

**Issue by arik-so** [CSSEGISandData/COVID-19#698]
In today's most recent update (commit a6681b6826f5f4cac4ae9caf998fd54ea8c7f4b9), the dataset for yesterday still has the same issues as before, where Italy, France, and a bunch of other countries have no changes from the day before.

**Comment by showdownhero** [comment]
Issues in Australia as well for 3/11 and 3/12 both days have identical data for all locations in Australia which I know is inaccurate

**Comment by arik-so** [comment]
Exactly. It's discussed extensively in #599.

**Comment by mklwong** [comment]
> 
> 
> Issues in Australia as well. For 3/11 and 3/12 both days have identical data for all locations in Australia which I know is inaccurate

I've created pull request #700 to fix the data for the 12th.

**Comment by emilianbold** [comment]
Probably related to #650, all recent data is flawed.

**Comment by JiPiBi** [comment]
The same for France : values for  13rd seem correct , but 12th is equal to 11th.

**Comment by r-lomba** [comment]
For those using Pandas, I propose the following fix for the 3/12 problem. Sources are cited. Not all the fixes, but the vast majority of the missing quantities are fixed here.

This allows you to:
- Pull the original CSVs from the CSSE source programatically, without worrying about having to fix them manually each time
- After download in Pandas Dataframe, apply the below fix in your code and 3/12 issue is fixed
- Proceed with your analysis further down in your code :)
- Disclaimer: maybe the Date format I used won't fit everybody's need. Adjust accordingly!

HTH

`#############################################################
##
## HARDCODED DICTIONARIES - FIXES TO MISSING DATA ON THE ORIGINAL
## DATASOURCE THAT NEVER GOT CORRECTED
##
## SOURCES:
## https://www.worldometers.info/coronavirus/
## https://ourworldindata.org/coronavirus-source-data
##
#############################################################

confirmed_fixes_dict = {'Italy|2020-03-12': 15113,
                        'Spain|2020-03-12': 3146,
                        'France|2020-03-12': 2876,
                        'United Kingdom|2020-03-12': 590,
                        'Germany|2020-03-12': 2745,
                        'Argentina|2020-03-12': 19,
                        'Australia|2020-03-12': 122,
                        'Belgium|2020-03-12': 314,
                        'Chile|2020-03-12': 23,
                        'Colombia|2020-03-12': 9,
                        'Greece|2020-03-12': 98,
                        'Indonesia|2020-03-12': 34,
                        'Ireland|2020-03-12': 43,
                        'Japan|2020-03-12': 620,
                        'Netherlands|2020-03-12': 503,
                        'Qatar|2020-03-12': 262,
                        'Singapore|2020-03-12': 178,
                        'Switzerland|2020-03-12': 645}
                        
deaths_fixes_dict = {'Italy|2020-03-12': 1016,
                     'Spain|2020-03-12': 86,
                     'France|2020-03-12': 61,
                     'United Kingdom|2020-03-12': 10,
                     'Germany|2020-03-12': 6,
                     'Argentina|2020-03-12': 1,
                     'Australia|2020-03-12': 3,
                     'Greece|2020-03-12': 1,
                     'Indonesia|2020-03-12': 1,
                     'Ireland|2020-03-12': 1,
                     'Japan|2020-03-12': 15,
                     'Netherlands|2020-03-12': 5,
                     'Switzerland|2020-03-12': 4}
                     
recovered_fixes_dict = {'Italy|2020-03-12': 1258,
                        'Spain|2020-03-12': 189,
                        'France|2020-03-12': 12,
                        'United Kingdom|2020-03-12': 19,
                        'Germany|2020-03-12': 25}
        

for key in confirmed_fixes_dict.keys():
    country_to_be_fixed = key.split('|')[0]
    date_to_be_fixed = key.split('|')[1]
    value_to_be_fixed = fixes_dict[key]
    dataframe_confirmed_DF.at[country_to_be_fixed, date_to_be_fixed] = value_to_be_fixed
        

for key in deaths_fixes_dict.keys():
    country_to_be_fixed = key.split('|')[0]
    date_to_be_fixed = key.split('|')[1]
    value_to_be_fixed = fixes_dict[key]
    dataframe_deaths_DF.at[country_to_be_fixed, date_to_be_fixed] = value_to_be_fixed
        

for key in recovered_fixes_dict.keys():
    country_to_be_fixed = key.split('|')[0]
    date_to_be_fixed = key.split('|')[1]
    value_to_be_fixed = fixes_dict[key]
    dataframe_recovered_DF.at[country_to_be_fixed, date_to_be_fixed] = value_to_be_fixed`

**Comment by iandoug** [comment]
Thanks.

**Comment by iandoug** [comment]
@r-lomba 


Mmm... just run the data. Am getting -7 new cases for 12 March for Switzerland. Assume due to different day-ends between WorldOMeter and JH?

**Comment by r-lomba** [comment]
@iandoug  you are right. The total confirmed for Switzerland was reported at 652 on JH on thata day. I had noticed no growth, and searched for the "real" number on the sources I listed, and I found a 645 which is obviously wrong...

I didn't get it, my bad. I have edited my message above and removed Switzerland.

Now I have also seen UK recoveries went -2 for that day too. Removed that too

Sorry for the mistakes, still believe the idea is good as it gives you a handy dictionary that can easily be maintained and updated without changing anything else... :)

**Comment by iandoug** [comment]
JH should have the data to fix this properly. In truth, someone like WHO should be publishing .csv files every hour, not relying on a distant university to do their job for them.

**Comment by emilianbold** [comment]
@iandoug WHO data is centralized into downloadable CSVs here https://ourworldindata.org/coronavirus-source-data but they don't track recovered cases.

**Comment by iandoug** [comment]
> @iandoug WHO data is centralized into downloadable CSVs here https://ourworldindata.org/coronavirus-source-data but they don't track recovered cases.

Yes thanks, I followed the link above. Also only updated once a day, and rather behind in their tally compared to WorldOMeter. WHO should be doing a better job here.

**Comment by santiagocasas** [comment]
@r-lomba  thanks for the Pandas fiix!
